### DISCUSSION TITLE
求助，Apple TV 使用Mitm无法播放。

### DISCUSSION THREAD

**Discussion by ParsonYuan** [DualSubs/.github#6] [Q&A 问答] [answered]
已将play.itunes.apple.com加入MitM列表，但仍无法播放，关闭Mitm可以正常播放。求助！

**Comment by ParsonYuan** [comment]
已解决，关闭iRingo Apple TV模块可恢复。

**Comment by ParsonYuan** [comment]
电脑端解决了，Apple TV和Ipad还是无法打开视频。已经正确安装证书。求助！谢谢

**Comment by VirgilClyne** [comment]
已经不需要在 `MitM` 中添加 `play.itunes.apple.com` 了

**Reply by ParsonYuan** [reply]
已删除play.itunes.apple.com，相同设置下（打开Mitm），Mac端可以打开且双语，Mac端Surge代理网络全部接管，但是Iphone/apple TV/Ipad都无法打开 苹果视频（🍎TV），报错（载入次内容时出现问题），关闭Mitm才可以播放。

**Reply by VirgilClyne** [reply]
1.  `🍿️ DualSubs: 🔣 Universal` 会 `MitM` `play-edge.itunes.apple.com` 以用来为 TV 添加双语字幕。
2. TV app `待播清单 (Up Next)` 的影片链接，因为本地缓存的影响，不会被立刻修复，如需播放，请长按影片图片，选择「前往单集」或「前往节目」从影片详情页面开始播放。
3. TV app `待播清单 (Up Next)` 的播放链接修复需要保持开启 ` iRingo: 📺 TV app` 模块的情况下，等待待播清单 (Up Next) 列表全部刷新一遍。
4.  并非所有 TV 中购买与租借影片都具有英文软字幕 (webVTT\TTML) 用来生成双语字幕，很大一部分购买与租借影片的英文字幕都是 CC 字幕(隐藏式字幕)，内封于视频流中，无法用来生成双语字幕

**Reply by ParsonYuan** [reply]
使用Mac电脑Surge app代理局域网络，打开Mitm，且各设备均正确信任同一证书。
1.经测试 iRingo模块打开，本机电脑和其他设备均无法播放TV 。
2.Universal模块打开，  iRingo模块关闭，本机电脑可以播放，且正常双语，其他设备均无法播放，报错“载入此内容时出现问题”
3.Universal模块和  iRingo模块均关闭，其他设备可正常播放，无双语。
暂未找到解决办法--！

**Reply by cynicalight** [reply]
> 使用Mac电脑Surge app代理局域网络，打开Mitm，且各设备均正确信任同一证书。 1.经测试 iRingo模块打开，本机电脑和其他设备均无法播放TV 。 2.Universal模块打开，  iRingo模块关闭，本机电脑可以播放，且正常双语，其他设备均无法播放，报错“载入此内容时出现问题” 3.Universal模块和  iRingo模块均关闭，其他设备可正常播放，无双语。 暂未找到解决办法--！

同样问题

**Reply by ParsonYuan** [reply]
目前其他使用均无问题，手机安装Surge iOS 打开Icloud同步设置，按要求安装模块，出现报错。反馈请求如下：
22:19:22.655424 [Connection] Handled by VIF
22:19:22.655596 [TLS] TLS Client Hello SNI: play-edge.itunes.apple.com
22:19:22.655629 [MITM] Enable MITM due to MITM hostname settings entry: play-edge.itunes.apple.com (🍿️ DualSubs: 🔣 Universal)
22:19:22.660231 [MITM] MITM failed. Client closed connection just after TLS handshake, it might because of certificate pinning.
期待解决。

**Reply by VirgilClyne** [reply]
> 目前其他使用均无问题，手机安装Surge iOS 打开Icloud同步设置，按要求安装模块，出现报错。反馈请求如下：
> 
> 22:19:22.655424 [Connection] Handled by VIF
> 
> 22:19:22.655596 [TLS] TLS Client Hello SNI: play-edge.itunes.apple.com
> 
> 22:19:22.655629 [MITM] Enable MITM due to MITM hostname settings entry: play-edge.itunes.apple.com (🍿️ DualSubs: 🔣 Universal)
> 
> 22:19:22.660231 [MITM] MITM failed. Client closed connection just after TLS handshake, it might because of certificate pinning.
> 
> 期待解决。

是 iOS 18.4 吗？

**Comment by li2282231** [comment]
你好，我也遇到这个问题了，请问最后怎么解决的？我是iPhone、Mac、Apple TV 4k三台设备，都安装了surge，且都正确配置了相同的证书，也都配置了Apple TV模块，目前iPhone和Mac可以正常使用Apple TV模块且配合双语字幕插件能正常显示双语字幕，但是Apple TV 4k则不行；如果Apple TV 4k配置了iringo的Apple TV模块及双语字幕插件后，首次播放是没问题的，但只要重启Apple TV 4k，再次进入Apple TV+ app就会报错，提示Apple TV app当前不可用，检查你的互联网连接，然后重试。

**Reply by ParsonYuan** [reply]
尝试退出TV端 apple账户，重新登陆；或者切换手机热点网络，进行重试。我有反复尝试，目前暂时可用

**Reply by li2282231** [reply]
我也发现了，当出现提示Apple TV app当前不可用的时候，退出TV端的Apple ID就可以登陆了，请问重新登陆后可否正常播放tv+的流媒体？是否仍有双语字幕？再一次重启TV端后打开Apple TV app还会提示Apple TV app当前不可用，检查你的互联网连接，然后重试吗？

**Comment by ParsonYuan** [comment]
重新登陆后，目前一切正常。可以双字幕显示。包括遥控关闭电源待机后重启。没试过断电后通电。



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: 李鹏洲 ***@***.***&gt;
发送时间: 2025年2月26日 23:55
抄送: ParsonYuan ***@***.***&gt;, Author ***@***.***&gt;
主题: Re: [DualSubs/.github] 求助，Apple TV 使用Mitm无法播放。 (Discussion #6)





 
我也发现了，当出现提示Apple TV app当前不可用的时候，退出TV端的Apple ID就可以登陆了，请问重新登陆后可否正常播放tv+的流媒体？是否仍有双语字幕？再一次重启TV端后打开Apple TV app还会提示Apple TV app当前不可用，检查你的互联网连接，然后重试吗？

**Reply by li2282231** [reply]
我刚测试了，断电后重启，包括系统里面重新启动tv端，Apple TV app就又会报错了；这是我提交的issue链接：
https://github.com/orgs/NSRingo/discussions/37#discussion-8018655
麻烦您移步看一下，是不是和您遇到的问题一样，谢谢！

**Reply by ParsonYuan** [reply]
并不具备解决的能力，期待大牛进一步解答！

**Reply by li2282231** [reply]
> 并不具备解决的能力，期待大牛进一步解答！

哈哈，我也是只会用，也善于发现问题，但是发现问题后就不知道该如何解决了！

**Reply by cynicalight** [reply]
> 重新登陆后，目前一切正常。可以双字幕显示。包括遥控关闭电源待机后重启。没试过断电后通电。 发自我的iPhone
> […](#)
> ------------------ 原始邮件 ------------------ 发件人: 李鹏洲 ***@***.***&gt; 发送时间: 2025年2月26日 23:55 抄送: ParsonYuan ***@***.***&gt;, Author ***@***.***&gt; 主题: Re: [DualSubs/.github] 求助，Apple TV 使用Mitm无法播放。 (Discussion #6) 我也发现了，当出现提示Apple TV app当前不可用的时候，退出TV端的Apple ID就可以登陆了，请问重新登陆后可否正常播放tv+的流媒体？是否仍有双语字幕？再一次重启TV端后打开Apple TV app还会提示Apple TV app当前不可用，检查你的互联网连接，然后重试吗？ — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

请问mac端是如何解决的，我是用的stash，配置了dual模块、信任证书、重新登录appleid，Mac端tv在点击播放时播放窗口在闪现之后退出了，tv主界面保持，好奇怪

**Reply by li2282231** [reply]
> > 重新登陆后，目前一切正常。可以双字幕显示。包括遥控关闭电源待机后重启。没试过断电后通电。 发自我的iPhone
> > […](#)
> > ------------------ 原始邮件 ------------------ 发件人: 李鹏洲 _**@**_._**> 发送时间: 2025年2月26日 23:55 抄送: ParsonYuan _**@**_.**_>, Author _**@**_._**> 主题: Re: [DualSubs/.github] 求助，Apple TV 使用Mitm无法播放。 (Discussion #6) 我也发现了，当出现提示Apple TV app当前不可用的时候，退出TV端的Apple ID就可以登陆了，请问重新登陆后可否正常播放tv+的流媒体？是否仍有双语字幕？再一次重启TV端后打开Apple TV app还会提示Apple TV app当前不可用，检查你的互联网连接，然后重试吗？ — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: _**@**_.**_>
> 
> 请问mac端是如何解决的，我是用的stash，配置了dual模块、信任证书、重新登录appleid，Mac端tv在点击播放时播放窗口在闪现之后退出了，tv主界面保持，好奇怪

你不能从主页上进入播放，必须要从前往单集或者前往节目后才能播放，这样就正常了。

**Reply by cynicalight** [reply]
需要开iringo吗，我是前往单集和前往节目都试过了，没有双语字幕的（没有任何多出来的字幕选项，正常情况应该会显示多个英语，其中两个是双语）。另外，我的配置是基于stash，不知道是否是stash的问题，感觉不是。

**Comment by li2282231** [comment]
当然要开启iringo的🍎TV模块啊，同时还有双语模块，二者都要开的，我没有stash，我用surge或者loon都是正常的。发自我的 iPhone在 2025年3月26日，20:56，bu44er ***@***.***> 写道：﻿
需要开iringo吗，我是前往单集和前往节目都试过了，没有双语字幕的（没有任何多出来的字幕选项，正常情况应该会显示多个英语，其中两个是双语）。另外，我的配置是基于stash，不知道是否是stash的问题，感觉不是。

**Reply by cynicalight** [reply]
好的 我再尝试一下 想问一下您的surge的设置 我的stash是规则模式+系统代理+增强模式+覆写的两个模块+mitm开启+证书信任

**Comment by ParsonYuan** [comment]
目前手机版本为18.3.2



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: 脆薯饼 ***@***.***&gt;
发送时间: 2025年3月30日 12:25
抄送: ParsonYuan ***@***.***&gt;, Author ***@***.***&gt;
主题: Re: [DualSubs/.github] 求助，Apple TV 使用Mitm无法播放。 (Discussion #6)





  
目前其他使用均无问题，手机安装Surge iOS 打开Icloud同步设置，按要求安装模块，出现报错。反馈请求如下：
 
22:19:22.655424 [Connection] Handled by VIF
 
22:19:22.655596 [TLS] TLS Client Hello SNI: play-edge.itunes.apple.com
 
22:19:22.655629 [MITM] Enable MITM due to MITM hostname settings entry: play-edge.itunes.apple.com (🍿️ DualSubs: 🔣 Universal)
 
22:19:22.660231 [MITM] MITM failed. Client closed connection just after TLS handshake, it might because of certificate pinning.
 
期待解决。
  
是 iOS 18.4 吗？

**Comment by mike8689** [comment]
surge 我在Mac装的可用，但是手机连接到电脑网关ios手机不能用 ，手机已经安装证书，Spotify手机版直接没有网络，

**Comment by VirgilClyne** [comment, marked as answer]
已在 [` iRingo: 📺 TV` v3.4.4](https://github.com/NSRingo/TV/releases/tag/v3.4.4) 和 [`🍿️ DualSubs: 🔣 Universal` v1.7.1](https://github.com/DualSubs/Universal/releases/tag/v1.7.1) 中修复
